tempesta_fw/http_msg.c

```diff
@@ -475,21 +475,56 @@ __hdr_sub(TfwHttpMsg *hm, char *name, size_t n_len, char *val, size_t v_len,
 			{ .ptr = name,	.len = n_len },
 			{ .ptr = ": ",	.len = 2 },
 			{ .ptr = val,	.len = v_len },
-			{ .ptr = "\r\n", .len = 2 }
+			{ .ptr = "\r\n",.len = 2 } /* may be adjusted */
 		},
 		.len = n_len + v_len + 4,
 		.flags = 4 << TFW_STR_CN_SHIFT
 	};
+	TfwStr *hlc = ((TfwStr *)hdr.ptr + 3); /* last @hdr chunk */
 
-	if (!TFW_STR_DUP(orig_hdr) && hdr.len <= orig_hdr->len) {
-		/* Rewrite the header in-place. */
-		if (ss_skb_cutoff_data(orig_hdr, hdr.len, 2))
-			return TFW_BLOCK;
-		if (tfw_strcpy(orig_hdr, &hdr))
+	if (TFW_STR_DUP(orig_hdr))
+		goto slow_path;
+
+	BUG_ON(!tfw_str_eolen(orig_hdr));
+
+	/*
+	 * We are trying to keep line endings as they were received, so
+	 * adjustment of @hdr's last chunk is needed in case of single LF.
+	 */
+	if (tfw_str_eolen(orig_hdr) == 1) {
+		--hdr.len;
+		++hlc->ptr;
+		--hlc->len;
+	}
```

Comment:
It seems ```hlc``` is used for length only, ```hlc->ptr``` isn't used. So can we just use int variable instead?